packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -198,3 +198,28 @@ export async function installCompilerWithUi(
   }
   return result;
 }
+
+export async function extractEmitterOptions(
+  baseDir: string,
+  packageName: string,
+): Promise<Record<string, any>> {
+  try {
+    const moduleResult = await loadModule(baseDir, packageName);
+    if (!moduleResult) {
+      logger.debug(`Failed to resolve module ${packageName}`);
+      return {};
+    }
+
+    const mainFilePath = moduleResult.type === "file" ? moduleResult.path : moduleResult.mainFile;
+    const moduleExports = await import(vscode.Uri.file(mainFilePath).toString());
+    if (moduleExports.$lib?.emitter?.options) {
+      return moduleExports.$lib.emitter.options;
+    }
+
+    logger.debug(`No emitter options schema found in ${packageName}`);
+    return {};
```

Comment:
return undefined for unexpected case, otherwise you can't distinct it from the emitter which does not have any options.